mysql-test/suite/rpl/t/rpl_warn.inc

```diff
@@ -0,0 +1,30 @@
+set timestamp=1000000000;
+CREATE TABLE t1(word VARCHAR(20));
+SHOW MASTER STATUS;
+--let position=query_get_value(SHOW MASTER STATUS, Position, 1)
+INSERT INTO t1 VALUES ("abirvalg");
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+SHOW MASTER STATUS;
+--let stop_position=query_get_value(SHOW MASTER STATUS, Position, 1)
+--let before_eof=`select $stop_position - 500`
+--let at_eof=`select $stop_position`
+--let after_eof=`select $stop_position + 500`
+--let MYSQLD_DATADIR= `select @@datadir;`
+
+--echo Case: Stop position before EOF
+--exec $MYSQL_BINLOG --short-form --start-position=$position --stop-position=$before_eof $MYSQLD_DATADIR/master-bin.000001 2>&1
+
+--echo Case: Stop position at EOF
+--exec $MYSQL_BINLOG --short-form --start-position=$position --stop-position=$at_eof $MYSQLD_DATADIR/master-bin.000001 2>&1
+
+--echo Case: Stop position after EOF
+--exec $MYSQL_BINLOG --short-form --start-position=$position --stop-position=$after_eof $MYSQLD_DATADIR/master-bin.000001 2>&1
```

Comment:
Generally it is better to avoid writing event output information to stdout (i.e. the `.result` file) if we can. Since this test isn't really about the event data, you can use the `--result-file` option to direct stdout to the file (and keep `2>&1` so the error output goes there too). Then you can assert that the warning is (or is not) in the file as intended, e.g. using `--source include/assert_grep.inc` (my preferred way, as it is more intentional) or `--source include/search_pattern_in_file.inc` (which just outputs the number of times the pattern exists).  A few bonuses of this too, are that you don't need 1) three different tests, as then they won't be `binlog_format` dependent anymore (you'll need to remove the `SHOW MASTER STATUS` too) , nor 2) the `set timestamp` (though kudos for figuring out those patterns :) ) 

Note too, an example you can follow for method 1 is in `rpl_xa_empty_transaction.test` for test case `33921.3`.

And while there, can you also add one more pattern from that file, to `--echo` the `MYSQL_BINLOG` command before the `--exec`?

Reply:
Very nice advice, I will implement it.

Reply:
In testing, the combined stdout and stderr don't end up in `--result-file`, but the warning does appear in the test's `.result` file.

Reply:
I updated the PR to include the changes for `--stop-position` with the thought that I would submit a separate PR for `--stop-datetime`.  From a software archaeology perspective, it would be cleaner because the changes and tests will standalone from those for `--stop-position`.  Will that work?

Reply:
I'm all for a cleaner software history. Though I think the `--stop-position` and `--stop-datetime` changes should be pushed at the same time for a consistent code base. What about using the same PR but have a separate commit?

Reply:
Sounds good!  I'm most interested in keeping them in separate commits; using the same PR works great for that 👍